include/dts-functions.sh

```diff
@@ -747,62 +747,34 @@ check_se_creds() {
   fi
 }
 
+semver() {
+  # return 1 if $1 < $2, 0 if $1 = $2 or -1 if $1 > $2
+
+  local valid1=
+  local valid2=
+  valid1="$(python3 -c "import semver; print(semver.Version.is_valid('$1'))")"
+  valid2="$(python3 -c "import semver; print(semver.Version.is_valid('$2'))")"
+  if [ "$valid1" != "True" ] || [ "$valid2" != "True" ]; then
+    error_exit "Incorrect version format"
+  fi
+  compare="$(python3 -c "import semver; print(semver.compare('$2', '$1'))")"
+  return "$compare"
+}
+
 compare_versions() {
     # return 1 if ver2 > ver1
     # return 0 otherwise
 
-    # Use awk to drop any suffixes (-devX or -rcY) from the version strings, so
-    # that only the major, minor, and patch versions are compared. If the
-    # resulting versions are the same (ver1 equals ver2), then the one without
-    # any suffix is considered to be the newer version.
-    local ver1 ver2 suffix1 suffix2
-    ver1="$(echo $1 | awk -F '-' '{print $1}')"
-    ver2="$(echo $2 | awk -F '-' '{print $1}')"
-    suffix1="$(echo $1 | sed 's/^[0-9]*\.[0-9]*\.[0-9]*-*//')"
-    suffix2="$(echo $2 | sed 's/^[0-9]*\.[0-9]*\.[0-9]*-*//')"
-
-    if [[ $ver1 =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] && [[ $ver2 =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-      IFS='.' read -r -a arr_ver1 <<< "$ver1"
-      IFS='.' read -r -a arr_ver2 <<< "$ver2"
-
-      if [ ${arr_ver2[0]} -lt ${arr_ver1[0]} ]; then
-          return 0
-      fi
-
-      if [ ${arr_ver2[0]} -gt ${arr_ver1[0]} ]; then
-          return 1
-      fi
-
-      if [ ${arr_ver2[0]} -eq ${arr_ver1[0]} ]; then
-          if [ ${arr_ver2[1]} -lt ${arr_ver1[1]} ]; then
-              return 0
-          fi
-
-          if [ ${arr_ver2[1]} -gt ${arr_ver1[1]} ]; then
-              return 1
-          fi
-
-          if [ ${arr_ver2[1]} -eq ${arr_ver1[1]} ]; then
-              if [ ${arr_ver2[2]} -lt ${arr_ver1[2]} ]; then
-                return 0
-              fi
-
-              if [ ${arr_ver2[2]} -gt ${arr_ver1[2]} ]; then
-                return 1
-              fi
-
-              # check suffixes
-              if [ ! -z "$suffix2" ] && [ -z "$suffix1" ]; then
-                return 0
-              fi
-
-              if [ -z "$suffix2" ] && [ ! -z "$suffix1" ]; then
-                return 1
-              fi
-          fi
-      fi
+    local ver1=
+    local ver2=
+    # convert version ending with`-rc<x>` to `-rc.<x>` where <x> is number
```

Comment:
please explain more why it is being done, maybe link semver spec as well

Reply:
Expanded comment in cacba6ee9673989b3df7106cd3614261ab6f16a3
Added information that this part is treated as alphanumeric identifier and compared like that and added link to https://semver.org